File: Gestion_Evenement/src/main/java/edu/esprit/entities/Document.java
package edu.esprit.entities;
import java.sql.Date;
public class Document {
    int id;
    String  titre;
Type type;
String url;
Niveau niveau;
Date date;
    Matiere mat;

    public Document() {
    }

    public Document(int id) {
        this.id = id;
    }

    public Document(String titre, Type type, String url, Niveau niveau, Date date,Matiere mat) {
        this.titre = titre;
        this.type = type;
        this.url = url;
        this.niveau = niveau;
        this.date = date;
        this.mat=mat;
    }

    public Document(int id, String titre, Type type, String url, Niveau niveau, Date date,Matiere mat) {
        this.id = id;
        this.titre = titre;
        this.type = type;
        this.url = url;
        this.niveau = niveau;
        this.date = date;
        this.mat=mat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Niveau getNiveau() {
        return niveau;
    }

    public void setNiveau(Niveau niveau) {
        this.niveau = niveau;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Matiere getMat() {
        return mat;
    }

    public void setMat(Matiere mat) {
        this.mat = mat;
    }

    @Override
    public String toString() {
        return "Document{" +
                "id=" + id +
                ", titre='" + titre + '\'' +
                ", type=" + type +
                ", url='" + url + '\'' +
                ", niveau=" + niveau +
                ", date=" + date +
                ", mat=" + mat +
                '}';
    }
}

File: projettest-main2/src/Entity/Document.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\DocumentRepository;
use Symfony\Component\Validator\Constraints\Date;

#[ORM\Entity(repositoryClass: DocumentRepository::class)]
class Document
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private  ?int $idDoc;
    #[ORM\Column(length: 255)]
    private  ?string $titre;

    #[ORM\Column(length: 255)]
    private ?string $type;

    #[ORM\Column(length: 1000)]
    private ?string $url;

    #[ORM\Column(length:255)]
    private ?string $niveau;

    #[ORM\Column]
    private ?Date $date;

    #[ORM\ManyToOne(targetEntity: Matiere::class, inversedBy: 'documents')]
    private ?Matiere $idMat;
    

    public function getIdDoc(): ?int
    {
        return $this->idDoc;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): static
    {
        $this->titre = $titre;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getUrl(): ?string
    {
        return $this->url;
    }

    public function setUrl(string $url): static
    {
        $this->url = $url;

        return $this;
    }

    public function getNiveau(): ?string
    {
        return $this->niveau;
    }

    public function setNiveau(string $niveau): static
    {
        $this->niveau = $niveau;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getIdMat(): ?Matiere
    {
        return $this->idMat;
    }

    public function setIdMat(?Matiere $idMat): static
    {
        $this->idMat = $idMat;

        return $this;
    }
    public function __toString()
    {
        return $this->getIdDoc();
    }



}